src/transaction.ts

```diff
@@ -0,0 +1,234 @@
+import { Psbt, networks } from 'liquidjs-lib';
+import { estimateTxSize, UtxoInterface } from './wallet';
+
+export interface OutputInterface {
+  value: number;
+  asset: string;
+  script: string;
+}
+
+export interface CoinSelectionResult {
+  selectedUtxos: UtxoInterface[];
+  changeOutputs: OutputInterface[];
+}
+
+// define a type using to implement changeScript strategy
+type ChangeScriptFromAssetGetter = (asset: string) => string | undefined;
+
+/**
+ * buildTx selects utxos among unspents to fill outputs' requirements,
+ * then it adds the associated inputs and outputs to the tx.
+ * finally it returns the new pset base64 encoded.
+ * @param psetBase64
+ * @param unspents
+ * @param outputs
+ * @param changeScriptByAsset
+ * @param addFee if true, estimate fee and add feeOutput to transaction (default = false)
+ * @param satsPerByte used for fee estimation (default = 0.1)
+ * @param network used for fee output (default = regtest)
+ */
+export function buildTx(
+  psetBase64: string,
+  unspents: UtxoInterface[],
+  outputs: OutputInterface[],
+  changeScriptByAsset: ChangeScriptFromAssetGetter,
+  addFee: boolean = false,
+  satsPerByte: number = 0.1,
+  network: networks.Network = networks.regtest
+): string {
+  const { selectedUtxos, changeOutputs } = greedyCoinSelection(
+    unspents,
+    outputs,
+    changeScriptByAsset
+  );
+
+  const inputs = selectedUtxos;
+
+  // if not fee, just add selected unspents as inputs and specified outputs + change outputs to pset
+  if (!addFee) {
+    const outs = outputs.concat(changeOutputs);
+    return addToTx(psetBase64, inputs, outs);
+  }
+
+  // otherwise, handle the fee output
+  const fee = createFeeOutput(
+    inputs.length,
+    outputs.length,
+    satsPerByte,
+    network
+  );
+
+  const changeIndexLBTC: number = changeOutputs.findIndex(
+    out => out.asset === network.assetHash
+  );
+  const diff = changeOutputs[changeIndexLBTC].value - fee.value;
```

Comment:
This is not safe so far.
I'm a dumb user and I provide only USDT unspents and outputs, but I want also to add fees because I'm really dumb. This method should return an error telling me how dumb I am, but instead it would throw an error, precisely at this line.

Since it seems that we need to add some validations for the arguments, I would also add a check that makes sure that if the addFee flag is set, then there must be some LBTCs within the provided unspents.

That said, even if the addFee flag is set, it could be possible that there's no LBTC change in the tx at this point. 

I think the overall logic of the fee output addition might be revised.. but let's move to offline for discussing about this.